service/storage_service.cc

```diff
@@ -619,14 +619,11 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
 }
 
 future<> storage_service::mark_existing_views_as_built(sharded<db::system_distributed_keyspace>& sys_dist_ks) {
-    return _db.invoke_on(0, [this, &sys_dist_ks] (replica::database& db) {
-        return do_with(db.get_views(), [this, &sys_dist_ks] (std::vector<view_ptr>& views) {
-            return parallel_for_each(views, [this, &sys_dist_ks] (view_ptr& view) {
-                return db::system_keyspace::mark_view_as_built(view->ks_name(), view->cf_name()).then([this, view, &sys_dist_ks] {
-                    return sys_dist_ks.local().finish_view_build(view->ks_name(), view->cf_name());
-                });
-            });
-        });
+    assert(this_shard_id() == 0);
```

Comment:
How do we know this is true? Previously the function would explicitly call shard 0, so I expect it was done for a reason - and `mark_existing_views_as_built` could have been called from other shards.

If this is really true, it should be explained in the commit message too, and `mark_existing_views_as_built` should get a comment at declaration site that it can be called only on shard 0.

Reply:
> How do we know this is true?

By ... reading the code. It's only called by `join_token_ring` which's also shard-0 only and even has its own assertion for this

> so I expect it was done for a reason

Added by a73d9ccf31b59f06232c112204eb5c7a3c427cc2 and this (slightly) surprises me too. Maybe the plan was that join-ring could happen _and_ _even_ _work_ on any shard

> mark_existing_views_as_built should get a comment at declaration site that it can be called only on shard 0.

sure thing